readthedocs/core/resolver.py

```diff
@@ -135,7 +139,9 @@ def resolve_domain(project, private=None):
     relation = project.superprojects.first()
     subdomain = getattr(settings, 'USE_SUBDOMAIN', False)
     prod_domain = getattr(settings, 'PRODUCTION_DOMAIN')
-    public_domain = getattr(settings, 'PUBLIC_DOMAIN', prod_domain)
+    public_domain = getattr(settings, 'PUBLIC_DOMAIN', None)
+    if public_domain is None:
+        public_domain = prod_domain
```

Comment:
`getattr` doesn't check for `None`, it only avoids `AttributeError` -- ie: https://gist.github.com/agjohnson/d97b9c4dbe9e3f1f0f42